enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,548 @@
+6. Insights Operators in connected clusters download the new remote configuration 
+   and start collecting the new data.
+7. The data requester makes use of the new data in newly incoming Insights Operator archives.
+
+### API Extensions
+
+This proposal does not add any API extension.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal does not require any specific details to account for the HyperShift. 
+For now, we can simply extend the remote configuration file to 
+request gathering of specific HyperShift resources.
+
+#### Standalone Clusters
+
+This proposal is relevant to standalone (and especially connected) clusters. 
+Standalone cluster retrieves the remote configuration (based on its OCP version) and collects data accordingly. 
+
+#### Single-node Deployments or MicroShift
+
+This proposal may lead to increased resource consumption by the Insights Operator 
+(described in the [Excessive use of cluster resources](#excessive-use-of-cluster-resources) section), 
+but the operator can be disabled in these environments or in case of problems. 
+
+### Implementation Details/Notes/Constraints 
+
+#### Remote Configuration Service
+
+The Insights Operator configuration will replace the current [conditional gathering endpoint](../insights/conditional-data-gathering.md)
+
+```bash
+https://console.redhat.com/api/gathering/gathering_rules.json
+```
+
+with a new one:
+
+```bash
+https://console.redhat.com/api/gathering/v2/%s/gathering_rules.json
+```
+
+The `%s` parameter is the OCP version of the Insights Operator as set in the `.status.versions` map 
+of the Insights `clusteoperator.config.openshift.io` resource (consider partially upgraded clusters).
+The remote configuration service will be responsible for serving configuration that can be 
+successfully processed by an Insights Operator of the given version.
+
+The new endpoint will return both the current conditional gathering configuration and the new remote
+configuration for container logs gathering. 
+In future versions, the endpoint may also include remote configuration for node logs and API resources.
+
+#### Remote Configuration Processing
+
+The Insights Operator will try to download the remote configuration before creating each archive (every 2 hours).
+This enables fast recovery time in case a remote configuration change caused issues and had to be reverted.
+The frequency is the same as with the 
+[conditional gathering configuration](../insights/conditional-data-gathering.md) (since OCP 4.10). 
+40k connected clusters requesting a remote configuration update every two hours mean 11 requests per second on average.
+
+It will validate the remote configuration using a hardcoded JSON schema 
+(it is part of the Insights Operator GitHub repository).
+The schema can evolve over time; we intend to change it only between x-stream and y-stream versions 
+(no backports) to simplify reasoning about differences between data gathered 
```

Comment:
This hasn't aged well under customer pressure in the past.  Especially as releases are supported for long. I suggest making sure you can accommodate backporting even if it isn't preferred.

Reply:
Thanks. Agreed.

Avoiding JSON schema changes in z-stream versions is only a preference. The design allows such changes at the cost of working with discontinuous version intervals in the remote configuration service and supporting tools. I've added a note in that sense.

Note that even if we backport a change, we will still need to support clusters without that change. The remote configuration service will need to serve a safe configuration to those clusters. Handling issues in the remote configuration service should reduce the need for backports.

Reply:
sounds good